### DISCUSSION TITLE
Node 18 - What adaptor is recommended?

### DISCUSSION THREAD

**Discussion by ShahSheel** [keycloak/keycloak#29026] [Adapter deprecation]
Hi all,

I'm currently upgrading our application from Node 12 -> Node 16 (Stable and seems to be working) to now Node 18. Please note, the current dependency were using is `4.0.0-beta.1`. Im fully aware how significantly outdated this is.  When doing a soft upgrade to node 18 it no longer works when connecting to keycloak so I assume theres been some changes from Node 16 -> 18 that effect this. 

Seeing that this dependency is going to be depreciated soon, does anyone recommend any alternatives to this?

**Comment by codespearhead** [comment]
[node-openid-client](https://github.com/panva/node-openid-client) is currently one of only two actively maintained, [certified](https://openid.net/developers/certified-openid-connect-implementations#:~:text=Certified%20Relying%20Party%20Libraries) OIDC-compliant Node clients available.

Duplicate: #9818 .

**Comment by ShahSheel** [comment]
Thanks for linking the duplicate number 

Closing this one
